arlas-tests/src/test/java/io/arlas/server/rest/plugins/eo/TileServiceIT.java

```diff
@@ -0,0 +1,142 @@
+    @Override
+    protected String getUrlPath(String collection) {
+        return arlasPath + "explore/" + collection + "/_tile";
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws JsonProcessingException, InterruptedException {
+        AbstractTestWithCollection.beforeClass();
+
+        ObjectMapper mapper = new ObjectMapper();
+        Data data = new Data();
+        int i=0,j=10;
+        data.id = String.valueOf("ID_" + i + "_" + j + "DI_bottom").replace("-", "_");
+        data.fullname = "My name is " + data.id;
+        data.params.age = Math.abs(i * j);
+        data.params.startdate = 1l * (i + 1000) * (j + 1000);
+        data.params.stopdate = 1l * (i + 1000) * (j + 1000) + 100;
+        data.geo_params.centroid = j + "," + i;
+        List<LngLatAlt> coords = new ArrayList<>();
+        coords.add(new LngLatAlt(i - 1, j + 1));
+        coords.add(new LngLatAlt(i + 1, j + 1));
+        coords.add(new LngLatAlt(i + 1, j - 1));
+        coords.add(new LngLatAlt(i - 1, j - 1));
+        coords.add(new LngLatAlt(i - 1, j + 1));
+        data.geo_params.geometry = new Polygon(coords);
+        String indexName=DataSetTool.ALIASED_COLLECTION?DataSetTool.DATASET_INDEX_NAME+"_original":DataSetTool.DATASET_INDEX_NAME;
+        DataSetTool.client.prepareIndex(indexName, DataSetTool.DATASET_TYPE_NAME, "ES_ID_TEST" + data.id)
+                .setSource(mapper.writer().writeValueAsString(data), XContentType.JSON)
+                .get();
+
+        data.id = String.valueOf("ID_" + i + "_" + j + "DI_top").replace("-", "_");
+        data.fullname = "My name is " + data.id;
+        DataSetTool.client.prepareIndex(indexName, DataSetTool.DATASET_TYPE_NAME, "ES_ID_TEST" + data.id)
+                .setSource(mapper.writer().writeValueAsString(data), XContentType.JSON)
+                .get();
+        Thread.sleep(5000);
+    }
+
+    @AfterClass
+    public static void afterClass() throws IOException {
+        AbstractTestWithCollection.afterClass();
+    }
+
+    @Test
+    public void testEmptyTopPyramid() throws InterruptedException {
+        // The top of the tile pyramid is empty
+        Stream.iterate(0, n -> n + 1).limit(DataSetTool.DATASET_TILE_URL.min_z)
+                .map(level ->
+                        given()
+                                .when()
+                                .get(getUrlPath(CollectionTool.COLLECTION_NAME) + "/" + level + "/1/1.png")
+                                .then()
+                                .statusCode(Response.Status.NO_CONTENT.getStatusCode())
+                );
+    }
+
+    @Test
+    public void testTile() throws InterruptedException, IOException {
+        // The top of the tile pyramid is empty => Coverage is around 50%
+        int coverage = ImageUtil.coverage(ImageIO.read(given()
+                .when()
+                .get(getUrlPath(CollectionTool.COLLECTION_NAME) + "/10/511/484.png?f=id:eq:ID_0_10DI_bottom")
```

Comment:
PNG files are tests resources but are located in `main` resources directory. As a result, they will be packaged in ARLAS-server shaded jar 😞

Reply:
This is necessary for the time being. A docker container will be in the near future available for providing those files.

Reply:
Ok